### ISSUE TITLE
About the Dictionary Learning 

### ISSUE THREAD

**Issue by TDBaker2016** [minhnhat93/lfa_sc#2]
Hi, minhnhat93. I tested your implementation of the LISTA. During the dictionary learning part, I find some bugs. The dictionary couldn't be learned when I tried to fix them. Could you please help me with it?

Besides, according to the original paper of LISTA in parameter setting, the learning rate for LISTA training decays as 1/n (n-iter number), but the thing is when I did the exact same thing. The gradient is exploded to NaN. I guess it is the single-sample training strategy instead of mini-batch training causes this, making the gradient unstable. I notice you set extremely small learning rate, so there is no such problem. My question is if it is possible to deploy batch gradient descends to address this problem?

Looking forword to hearing from you!

**Comment by minhnhat93** [comment]
Hi TDBaker2016,

I think batch training could work and would work way better. This
Github repository is what I did almost 2 years before when I knew almost
nothing about training neural networks and I did the gradient descent from
scratch so there might be a lot of bugs in it. It would be the best for you
to consider using one of the other frameworks like Tensorflow with nicer
interfaces and automatic gradient computing. The repository is no longer
maintained and I moved to new projects. Sorry.

*Minh Nhat Nguyen*
Grad Student,
Department of Computing Science, University of Alberta
Email: minhnhat93@gmail.com / nmnguyen@ualberta.ca
